src/hotspot/share/runtime/os.inline.hpp

```diff
@@ -27,6 +27,37 @@
 
 #include "runtime/os.hpp"
 
+#include OS_HEADER(os)
 #include OS_HEADER_INLINE(os)
+#include OS_CPU_HEADER_INLINE(os)
+
+// Below are inline functions that are rarely implemented by the platforms.
+// Provide default emptyy implementation.
+
+#ifndef PLATFORM_PRINT_NATIVE_STACK
+inline bool os::platform_print_native_stack(outputStream* st, const void* context,
+                                     char *buf, int buf_size) {
+  return false;
+}
+#endif
+
+#ifndef HAVE_CDS_CORE_REGION_ALIGNMENT
+inline size_t os::cds_core_region_alignment() {
+  return (size_t)os::vm_allocation_granularity();
+}
+#endif
+
+#ifndef _WINDOWS
+// Currently used only on Windows.
+inline bool os::register_code_area(char *low, char *high) {
+  return true;
+}
+#endif
+
+#ifndef HAVE_FUNCTION_DESCRIPTORS
+inline void* os::resolve_function_descriptor(void* p) {
+  return NULL;
+}
```

Comment:
This is unnecessary and could be removed if did:

```
+#ifdef HAVE_FUNCTION_DESCRIPTORS
   // Used only on PPC.
   inline static void* resolve_function_descriptor(void* p);
+#endif
```

in `os.hpp`

Reply:
After this PR, os.hpp no longer includes any os- or cpu-specific os_xxx.hpp files. I did this to avoid including large amount of os-specific declarations (such as os::Linux, which is about 400 lines) into generic source files, which aren't interested in these declarations.

As a result of this change, there's no way to let os.hpp know whether `HAVE_FUNCTION_DESCRIPTORS` is defined for the current OS+CPU. If we want to do that, we need to either:

- include os_xxx.hpp back in os.hpp (which I want to avoid)
- add new header files such as `os_<os>.defs.hpp`  `os_<os>_<cpu>.defs.hpp` to define  `HAVE_FUNCTION_DESCRIPTORS`  (this will result in lots of header files that usually have nothing inside)

Since there are only 4 functions that fall under this pattern (functions that are used in generic code but most platforms have dummy implementations), I think what I have is a good compromise

Reply:
Could we do:

```
#if defined(LINUX) && defined(PPC) && !defined(ABI_ELFv2)
  #define HAVE_FUNCTION_DESCRIPTORS
#endif
```

in os.hpp and whatever the equivalent for AIX/PPC is?

Reply:
```
#if defined(PPC) && !defined(ABI_ELFv2)
  #define HAVE_FUNCTION_DESCRIPTORS
#endif
```
should do the trick. AIX uses XCOFF, not Elf, so ABI_ELFv2 should never be defined.

But I have no emotions. We can leave it like this. Or do it in a follow up.

Reply:
```
#if defined(PPC) && !defined(ABI_ELFv2)
  #define HAVE_FUNCTION_DESCRIPTORS
#endif
```
should do the trick. AIX uses XCOFF, not Elf, so ABI_ELFv2 should never be defined.

But I have no emotions. We can leave it like this. Or do it in a follow up.